pkg/dns/aws/filewatcher.go

```diff
@@ -0,0 +1,76 @@
+package aws
+
+import (
+	"bytes"
+	"io/ioutil"
+	"time"
+
+	certsutil "github.com/openshift/cluster-ingress-operator/pkg/util/certs"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// startWatcher starts the FileWatcher and periodically ensures DNS clients
+// are using the current ca bundle until a message is received on the stop
+// or error channels.
+func (m *Provider) startWatcher(stop <-chan struct{}) error {
+	go wait.Until(func() {
+		if err := m.ensureDNSClientsTLSTransport(); err != nil {
+			log.Error(err, "failed to ensure dns client tls transport")
+		}
+	}, 1*time.Minute, stop)
+
+	errChan := make(chan error)
+	go func() {
+		errChan <- m.fileWatcher.Start(stop)
+	}()
+
+	// Wait for the watcher to exit or an explicit stop.
```

Comment:
this is a bit strange.  Why launch two goroutines and then have a dummy waiter here?  Can we just make `m.fileWatcher.Start` blocking, or is that not something we control.

Reply:
@wking I used the ingress operator's [start()](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/operator.go#L125-L153) as a reference. The first goroutine is periodically ensuring the ca bundle used by dns clients is up-to-date while the second goroutine  starts the file watcher.